### PR TITLE
add-etcd-cluster-migrator

### PR DESCRIPTION
**Pull request by calvix**
Install `etcd-cluster-migrator` app into cluster when the cluster is flagged as HA.

We decided to install it every time to avoid having a complex solution that would decide when it can be installed. The migrator checks for current status and does nothing in case the etcd cluster has three members.

the issue to sort this later is here: https://github.com/giantswarm/giantswarm/issues/11397

### CODE REVIEW

**Comment by calvix** [comment]
RFR - there will be a change to the chart to have there stable released version once all PRs for `etcd-cluster-migrator` are merged

But otherwise, this works.  When you upgrade cluster from 1 to 3 masters, this will automatically migrate etcd cluster and sync the data between them.

No manual steps are necessary,

File: service/internal/cloudconfig/tccpn_extension.go [hunk 1]
```diff
@@ -187,6 +189,41 @@ func (e *TCCPNExtension) Files() ([]k8scloudconfig.FileAsset, error) {
 		},
 	}
 
+	// We install etcd-cluster-migrator in every case of HA masters.
+	// etcd-cluster-migrator does not break  installation that was created as HA master or was already migrated.
+	// This is jsut workaround for now so we dont spent too much time thinking how to deploy
+	// etcd-cluster-migrator on demand.
+	// https://github.com/giantswarm/giantswarm/issues/11397
```
Comment (xh3b4sd) [inline service/internal/cloudconfig/tccpn_extension.go:196]:
```suggestion
	// TODO we install etcd-cluster-migrator in every case of HA masters. The etcd-cluster-migrator app 
	// does not have negative effects on Tenant Clusters that were already created using the HA masters 
	// setup. Already migrated Tenant Clusters can also safely run this app for the time being. The 
	// workaround here for now is only so we don't have to spent too much time implementing a proper
	// managed app via our app catalogue, which only deploys the etcd-cluster-migrator on demand in 
	// case a Tenant Cluster is migrating automatically from 1 to 3 masters. See also the TODO issue below.
	// 
	//     https://github.com/giantswarm/giantswarm/issues/11397
	//
```

File: service/internal/cloudconfig/tccpn_extension.go [hunk 2]
```diff
@@ -318,6 +356,21 @@ func (e *TCCPNExtension) Units() ([]k8scloudconfig.UnitAsset, error) {
 		},
 	}
 
+	// We install etcd-cluster-migrator in every case of HA masters.
+	// etcd-cluster-migrator does not break installation that was created
+	// as HA masters or was already migrated. This is just workaround
+	// for now so we dont spent too much time thinking how to deploy
+	// etcd-cluster-migrator on demand.
+	// https://github.com/giantswarm/giantswarm/issues/11397
```
Comment (xh3b4sd) [inline service/internal/cloudconfig/tccpn_extension.go:364]:
```suggestion
	// TODO we install etcd-cluster-migrator in every case of HA masters. The etcd-cluster-migrator app 
	// does not have negative effects on Tenant Clusters that were already created using the HA masters 
	// setup. Already migrated Tenant Clusters can also safely run this app for the time being. The 
	// workaround here for now is only so we don't have to spent too much time implementing a proper
	// managed app via our app catalogue, which only deploys the etcd-cluster-migrator on demand in 
	// case a Tenant Cluster is migrating automatically from 1 to 3 masters. See also the TODO issue below.
	// 
	//     https://github.com/giantswarm/giantswarm/issues/11397
	//
```

File: service/internal/cloudconfig/template/etcd_cluster_migrator.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
+package template
+
```
Comment (xh3b4sd) [inline service/internal/cloudconfig/template/etcd_cluster_migrator.go:2]:
```suggestion

	// TODO we install etcd-cluster-migrator in every case of HA masters. The etcd-cluster-migrator app 
	// does not have negative effects on Tenant Clusters that were already created using the HA masters 
	// setup. Already migrated Tenant Clusters can also safely run this app for the time being. The 
	// workaround here for now is only so we don't have to spent too much time implementing a proper
	// managed app via our app catalogue, which only deploys the etcd-cluster-migrator on demand in 
	// case a Tenant Cluster is migrating automatically from 1 to 3 masters. See also the TODO issue below.
	// 
	//     https://github.com/giantswarm/giantswarm/issues/11397
	//
```

File: service/internal/cloudconfig/template/etcd_cluster_migrator.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  namespace: giantswarm
+spec:
+  config:
+    configMap:
+      name: etcd-cluster-migrator-chart-values
+      namespace: giantswarm
+      resourceVersion: ""
+  name: etcd-cluster-migrator
+  namespace: kube-system
+  tarballURL: https://giantswarm.github.io/giantswarm-playground-test-catalog/etcd-cluster-migrator-0.0.0-07610a3b18c6fc5f9ec28d51a78f121974836c21.tgz
```
Comment (xh3b4sd) [inline service/internal/cloudconfig/template/etcd_cluster_migrator.go:66]: I guess this is the version we have to fix once the migrator app PRs are merged.

Comment (calvix) [reply service/internal/cloudconfig/template/etcd_cluster_migrator.go]: Yep

**Pull request merged**